modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v1_1_6/UpgradeDiscussionSubscriptionClassName.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.journal.internal.upgrade.v1_1_6;
+
+import com.liferay.comment.upgrade.BaseUpgradeDiscussionSubscriptionClassName;
+import com.liferay.journal.model.JournalArticle;
+import com.liferay.subscription.service.SubscriptionLocalService;
+
+/**
+ * @author Roberto Díaz
+ */
+public class UpgradeDiscussionSubscriptionClassName
+	extends BaseUpgradeDiscussionSubscriptionClassName {
+
+	public UpgradeDiscussionSubscriptionClassName(
+		SubscriptionLocalService service) {
+
+		subscriptionLocalService = service;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		addSubscriptions();
```

Comment:
Why not call `deleteSubscriptions` here? I see no reason why we should leave the ones with the wrong classname in the database.

Reply:
En este caso es posible suscribirse tanto a los comentarios como a la entrada en sí

Dado que no sabemos cómo se creó la suscripcion inicial, no podemos borrarla, solo duplicarla para ambos casos.